lib/Dialect/TritonNvidiaGPU/Transforms/OptimizeTMemLayouts.cpp

```diff
@@ -116,9 +116,16 @@ class TMemSplitLoadPattern : public OpRewritePattern<SplitOp> {
       Value subSlice = rewriter.create<TMEMSubSliceOp>(tmemLoad.getLoc(), tmem,
                                                        nOffset, splitNSize);
 
+      // FIXME: This does not seem right. It might be hiding a different bug
       // Choose a layout compatible with the slice size.
+      // Use the TMEM memdesc block size for N when selecting a compatible
+      // layout. The compatibility checker is defined with respect to the
+      // memdesc encoding, not the slice width.
+      auto tmemEnc = cast<triton::nvidia_gpu::TensorMemoryEncodingAttr>(
+          tmemLoad.getSrc().getType().getEncoding());
+      int blockN = tmemEnc.getBlockN();
```

Comment:
you're right it should be the size of blockN however we should pick the blockN of `subSlice` and not the original alloc

Reply:
Yeah, I'm not 100% sure what's going on here. It might be better to punt on this for now until we have a more solid layout inference of DistributedLayouts from TMEM layouts and then revisit this?